### DISCUSSION TITLE
Strange copy/paste behavior in Powershell 7

### DISCUSSION THREAD

**Discussion by mcdonamw** [PowerShell/PowerShell#20507] [General]
I'm trying to paste the following code into a PowerShell 7 window:
```
Get-VMHost | sort Name | ForEach-Object -Process { 
  [PSCustomObject]@{
	'lastpatchdate' = [datetime]((Get-ESXCli -VMHost $_).software.vib.list() | Select-Object -Property installdate -ExpandProperty installdate | Sort-Object -Descending)[0]
	'host'              = $_.Name
	'ipaddress'         = (Get-VMHostNetworkAdapter -VMHost $_ -VMKernel | ?{$_.ManagementTrafficEnabled}).IP -Join ";"
	'version'           = $_.version
	'build'             = $_.build
	'manufacturer'      = $_.manufacturer
	'model'             = $_.model
	'serialnumber'      = (Get-ESXCli -VMHost $_).hardware.platform.get().SerialNumber
	'cpu'               = $_.processortype
	'numcpu'            = $_.numcpu
	'hyperthreadactive' = $_.hyperthreadingactive
	'cputotalMhz'       = $_.cputotalmhz
	'memorytotalGB'     = [math]::ceiling($_.memorytotalgb)
  } 
}
```

But when it pastes, it pastes as follows causing the code to error:  ***Notice the unexpected text being prepended on some lines***

```
PS C:\Users\158088> Get-VMHost | sort Name | ForEach-Object -Process {
>>   [PSCustomObject]@{
>>     'lastpatchdate' = [datetime]((Get-ESXCli -VMHost $_).software.vib.list() |
>>       Select-Object -Property installdate -ExpandProperty installdate |
>>       Sort-Object -Descending)[0]
>>     'host'              = $_.Name
>> host'ipaddress'         = (Get-VMHostNetworkAdapter -VMHost $_ -VMKernel | ?{$_.ManagementTrafficEnabled}).IP
>>     'version'           = $_.version
>>     'build'             = $_.build
>> lastpatchdate'manufacturer'      = $_.manufacturer
>> lastpatchdate'model'             = $_.model
>> lastpatchdate'serialnumber'      = (Get-ESXCli -VMHost $_).hardware.platform.get().SerialNumber
>> lastpatchdate'cpu'               = $_.processortype
>> lastpatchdate'numcpu'            = $_.numcpu
>> lastpatchdate'hyperthreadactive' = $_.hyperthreadingactive
>> lastpatchdate'cputotalMhz'       = $_.cputotalmhz
>> lastpatchdate'memorytotalGB'     = $_.memorytotalgb
>> lastpatchdate
>>   }
>> }
```

Any idea what's going on here?  I tried a new powershell window but the same results.  This pastes fine in Windows Powershell 5.

**Comment by mcdonamw** [comment]
Well it looks like I figured it out.  Apparently if there are tabs being used to space content, it trips up the paste.  Surely that has to be a bug?  My code editor of choice, Notepad++, automatically adds tabs on new lines if I am indenting code.

**Comment by MartinGC94** [comment]
When you right click to paste into the console you are using the built-in pasting feature of the console, which types in the characters one by one. Because the "Tab" key is used to tabcomplete code it makes sense that PS sees the pasted tab character and tries to complete the code. If you use the Ctrl+v shortcut it's PSReadline that handles the pasting and it will correctly paste it in as one large block without any funky tab completion business.

The only curious thing here is that it's attempting to provide completions for the hashtable you are currently writing and the suggestions are based on keys you've already written. That should be fixed. Feel free to open an issue about that so it can be fixed.

**Reply by mcdonamw** [reply]
Ah!  That makes sense.  Thanks for the explanation!

**Reply by mcdonamw** [reply]
Issue #20824 has been created.
